storage-account.tf

```diff
@@ -1,5 +1,86 @@
 // Shared and specialised Storage Accounts
 
+locals {
+  mgmt_network_name    = "core-cftptl-intsvc-vnet"
+  mgmt_network_rg_name = "aks-infra-cftptl-intsvc-rg"
+  
+  sa_subnets = [
+    data.azurerm_subnet.jenkins_subnet.id,
+    data.azurerm_subnet.aks-00-mgmt.id,
+    data.azurerm_subnet.aks-01-mgmt.id,
+    data.azurerm_subnet.aks-00-infra.id,
+    data.azurerm_subnet.aks-01-infra.id,
+    data.azurerm_subnet.aks-00-preview.id,
+    data.azurerm_subnet.aks-01-preview.id
+  ]
+}
+
+data "azurerm_virtual_network" "mgmt_vnet" {
+  provider            = azurerm.mgmt
+  name                = local.mgmt_network_name
+  resource_group_name = local.mgmt_network_rg_name
+}
+
+data "azurerm_subnet" "jenkins_subnet" {
```

Comment:
would probably be better to extract a local var with the subnet names,

`local.jenkins_subnets = ["iaas", "aks-00", "aks-01"]`

and then use a `for_each` over the local

and then use something like https://www.terraform.io/docs/language/functions/concat.html to join the list you get